### PR TITLE
python3.7 in travis

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

**Comment by asottile** [comment]
I'm not 100% on whether this is the right way to `allow_failures` or what build stage you wanted python3.7 in -- I guess we'll see!

File: .travis.yml [hunk 1]
```diff
@@ -58,6 +58,9 @@ jobs:
     - env: TASK=check-py27-typing
     - env: TASK=check-py34
     - env: TASK=check-py35
+    - env: TASK=check-py37
+      sudo: required
```
Comment (alexwlchan) [inline .travis.yml]: I didn’t realise it was possible to specify `sudo: required` or `dist: xenial` on a per-task basis – I thought they applied to every build.

Comment (asottile) [reply .travis.yml]: yeah this has been a feature of `matrix` for a while, as for python3.7-specific research here's my current writeup:

https://github.com/deadsnakes/travis-ci-python3.7-example#travis-ci-python37-example

**Comment by asottile** [comment]
wow, my guess at `allow_failures` worked the first time too -- @DRMacIver this should be reviewable now :)

File: .travis.yml (hunk 1 above)
Comment (DRMacIver) [reply .travis.yml]: Yeah, this was super surprising to me too, and really makes the difference - a lot of my resistance to turning on `sudo: required` was that I thought it was needed for *all* jobs, not just individual ones.

**Pull request merged**